plugins/cuda/cuda_plugin.c

```diff
@@ -470,6 +471,18 @@ int cuda_plugin_resume_devices_late(int pid)
 }
 CR_PLUGIN_REGISTER_HOOK(CR_PLUGIN_HOOK__RESUME_DEVICES_LATE, cuda_plugin_resume_devices_late)
 
+/**
+ * The cuda-checkpoint tool requires the libcuda library.
```

Comment:
* does it really require it? If I remember right, cude-checkpoint initially was a static-linked binary. Did I miss something?

* The idea was to find out whether a host has a required hardware. For example, user can use one vm image for all their vm-s, but only some of them will have nvidia cards. This check will always pass even when a host doesn't have any nvidia cards.

Reply:
>does it really require it?

yes, the `cuda-checkpoint` tool is linked against `libcuda.so.1` and it fails if this library is not installed.

> This check will always pass even when a host doesn't have any nvidia cards.

We can also check the content of `/proc/driver/nvidia/gpus/*`:

```
# System without GPU
$ ls /proc/driver/nvidia/gpus/*
ls: cannot access '/proc/driver/nvidia/gpus/*': No such file or directory

# System with GPU
$ ls /proc/driver/nvidia/gpus/
0000:01:00.0

$ ls  /proc/driver/nvidia/gpus/
0000:82:00.0
```

Reply:
I've updated the pull request to check only for `/proc/driver/nvidia/gpus/*`.